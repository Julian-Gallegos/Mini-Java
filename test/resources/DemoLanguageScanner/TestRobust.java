/*
This is a simple class.
*/
public class Hello {
    public static void main(String[] args) {
        System.out.println("Hello!"); 

        Person[] people = new Person[3]; 
        people[0] = new Person("steve", 5, true);
        people[1] = new Person("john", 10, false);
        people[2] = new Person("justin", 3, true);
        
        // each person will say hello.
        for (int i = 0; i < people.length; i++) {
            Person p = people[i];
            int j = p.getRepeat() * 2; 
            System.out.println(p.getName() + "_says_hello!");
            while (j != 0) {
                if (p.isEmployed() && true) {
                    System.out.println("Where's_my_hazard_pay!");
                } else {
                    System.out.println("Where's_my_stimulus?");
                }
                j--;
            }
        }
    }
}

/*
extending object is redundant. 
*/ 
class Person extends Object {
    private String name; 
    private int repeat; 
    private boolean isEmployed; 

    Person(String name, int repeat, boolean isEmployed) {
        this.name = name; 
        this.repeat = repeat; 
        this.isEmployed = isEmployed;
    }

    int getRepeat() {
        return repeat; 
    }

    boolean isEmployed() {
        return isEmployed; 
    }

    String getName() {
        return name; 
    }
}