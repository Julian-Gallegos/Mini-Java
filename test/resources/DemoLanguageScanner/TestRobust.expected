PUBLIC CLASS ID(Hello) LBRACE PUBLIC STATIC VOID MAIN LPAREN ID(String) LBRACKET RBRACKET ID(args) RPAREN LBRACE SYSOUTPRINTLN LPAREN TEXT("Hello!") RPAREN SEMICOLON ID(Person) LBRACKET RBRACKET ID(people) BECOMES NEW ID(Person) LBR
ACKET DIGIT(3) RBRACKET SEMICOLON ID(people) LBRACKET DIGIT(0) RBRACKET BECOMES NEW ID(Person) LPAREN TEXT("steve") COMMA DIGIT(5) COMMA TRUE RPAREN SEMICOLON ID(people) LBRACKET DIGIT(1) RBRACKET BECOMES NEW ID(Person) LPAREN TEXT(
"john") COMMA DIGIT(10) COMMA FALSE RPAREN SEMICOLON ID(people) LBRACKET DIGIT(2) RBRACKET BECOMES NEW ID(Person) LPAREN TEXT("justin") COMMA DIGIT(3) COMMA TRUE RPAREN SEMICOLON ID(for) LPAREN INT ID(i) BECOMES DIGIT(0) SEMICOLON I
D(i) LTHAN ID(people) DOT LENGTH SEMICOLON ID(i) PLUS PLUS RPAREN LBRACE ID(Person) ID(p) BECOMES ID(people) LBRACKET ID(i) RBRACKET SEMICOLON INT ID(j) BECOMES ID(p) DOT ID(getRepeat) LPAREN RPAREN TIMES DIGIT(2) SEMICOLON SYSOUTPR
INTLN LPAREN ID(p) DOT ID(getName) LPAREN RPAREN PLUS TEXT(" says hello!") RPAREN SEMICOLON WHILE LPAREN ID(j) NOT BECOMES DIGIT(0) RPAREN LBRACE IF LPAREN ID(p) DOT ID(isEmployed) LPAREN RPAREN AND TRUE RPAREN LBRACE SYSOUTPRINTLN
LPAREN TEXT("Where's my hazard pay!") RPAREN SEMICOLON RBRACE ELSE LBRACE SYSOUTPRINTLN LPAREN TEXT("Where's my stimulus?") RPAREN SEMICOLON RBRACE ID(j) MINUS MINUS SEMICOLON RBRACE RBRACE RBRACE RBRACE CLASS ID(Person) EXTENDS ID(
Object) LBRACE ID(private) ID(String) ID(name) SEMICOLON ID(private) INT ID(repeat) SEMICOLON ID(private) BOOL ID(isEmployed) SEMICOLON ID(Person) LPAREN ID(String) ID(name) COMMA INT ID(repeat) COMMA BOOL ID(isEmployed) RPAREN LBRA
CE THIS DOT ID(name) BECOMES ID(name) SEMICOLON THIS DOT ID(repeat) BECOMES ID(repeat) SEMICOLON THIS DOT ID(isEmployed) BECOMES ID(isEmployed) SEMICOLON RBRACE INT ID(getRepeat) LPAREN RPAREN LBRACE RETURN ID(repeat) SEMICOLON RBRA
CE BOOL ID(isEmployed) LPAREN RPAREN LBRACE RETURN ID(isEmployed) SEMICOLON RBRACE ID(String) ID(getName) LPAREN RPAREN LBRACE RETURN ID(name) SEMICOLON RBRACE RBRACE